ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -606,6 +614,36 @@ getCompletions plugins ideOpts CC {allModNamesAsNS, anyQualCompls, unqualCompls,
                   hpos = upperRange position'
               in getCContext lpos pm <|> getCContext hpos pm
 
+
+          -- we need the hieast to be fresh
+          -- not fresh, hasfield won't have a chance. it would to another larger change to ghc IfaceTyCon to contain record fields
+          recordDotSyntaxCompls :: [(Bool, CompItem)]
+          recordDotSyntaxCompls = case maybe_ast_res of
+            Just (HAR {hieAst = hieast, hieKind = HieFresh},_) -> concat $ pointCommand hieast (completionPrefixPos prefixInfo) nodeCompletions
+            _ -> []
+            where
+              nodeCompletions :: HieAST Type -> [(Bool, CompItem)]
+              nodeCompletions node = concatMap g (nodeType $ nodeInfo node)
+              g :: Type -> [(Bool, CompItem)]
+              g (TyConApp theTyCon _) = map (dotFieldSelectorToCompl (printOutputable $ GHC.tyConName theTyCon)) $ getSels theTyCon
+              g _ = []
+              getSels :: GHC.TyCon -> [T.Text]
+              getSels tycon = let f fieldLabel = printOutputable fieldLabel
+                              in map f $ tyConFieldLabels tycon
+              dotFieldSelectorToCompl :: T.Text -> T.Text -> (Bool, CompItem)
+              --dotFieldSelectorToCompl label = (True, CI CiVariable label (ImportedFrom T.empty) Nothing label Nothing emptySpanDoc False Nothing)
+              dotFieldSelectorToCompl recname label = (True, CI
+                { compKind = CiField
+                , insertText = label
+                , provenance = DefinedIn recname
+                , typeText = Nothing
+                , label = label
+                , isInfix = Nothing
+                , docs = emptySpanDoc
```

Comment:
it should have doc, probably?

Reply:
I believe that other autocompletions don't provide docs atm, e.g. `Logic.hs:344`. Though we do provide them for snippets. I worry putting something like "record dot completion" would distract the developer. Did you have something else in mind?

Reply:
Well, fields have haddock usually, you know
```
data Foo {
  hello :: Int -- ^ This is the doc for foo
}
```
I would expect to get the field haddock in the completion doc!

This might get automatically sorted out for us with https://github.com/haskell/haskell-language-server/pull/3204, not sure.

Reply:
Ah that makes much more sense than what I was thinking. If that pr doesn't sort this out, it looks like it will provide the machinery to make it easier.